### DISCUSSION TITLE
Is codegen tool going to support std::future implementation for proxy async methods?

### DISCUSSION THREAD

**Discussion by ghost** [Kistler-Group/sdbus-cpp#317] [Q&A]
I've seen that std::future based async calls support has been added [recently](https://github.com/Kistler-Group/sdbus-cpp/pull/311/commits). However the code generator tool for proxies only generates the callback based implementation.

Is it planned to add this feature to the codegen as well in order to decide which option to use?

Thanks!

**Comment by sangelovic** [comment]
It was not part of my short term plan. But if you as a client desire that, we can implement it. Your contribution would be very welcome. Would you be willing to contribute here?

-------- Original Message --------

**Comment by sangelovic** [comment]
Implemented with https://github.com/Kistler-Group/sdbus-cpp/pull/353
